src/es_archiver/actions/rebuild_all.ts

```diff
@@ -58,11 +65,11 @@ export async function rebuildAllAction({ dataDir, log, rootDir = dataDir }) {
     const tempFile = childPath + (gzip ? '.rebuilding.gz' : '.rebuilding');
 
     await createPromiseFromStreams([
-      createReadStream(childPath),
+      createReadStream(childPath) as Readable,
       ...createParseArchiveStreams({ gzip }),
       ...createFormatArchiveStreams({ gzip }),
       createWriteStream(tempFile),
-    ]);
+    ] as [Readable, ...Writable[]]);
```

Comment:
(There are a lot of these in the PR, will only comment once). Typescript is not able to understand the `[Readable, ...Writable[]]` signature due to the fact that most transformers are `Duplex` ( both Readable and Writable). I had to help him.